pydm/widgets/spinbox.py

```diff
@@ -175,3 +201,79 @@ def showStepExponent(self, val):
         """
         self._show_step_exponent = val
         self.update_format_string()
+
+    @Property(bool)
+    def limitsFromChannel(self):
+        """
+        A choice whether or not to use the limits given by channel.
+
+        Returns
+        -------
+        limits_from_channel : bool
+            True means that the widget will use the limits information
+            from the Channel if available.
+        """
+        return self._limits_from_channel
+
+    @limitsFromChannel.setter
+    def limitsFromChannel(self, value):
+        """
+        A choice whether or not to use the limits given by channel.
+
+        Parameters
+        ----------
+        value : bool
+            True means that the widget will use the limits information
+            from the PV if available.
+        """
+        if self._limits_from_channel != bool(value):
+            self._limits_from_channel = bool(value)
+            self.update_limits()
+
+    @Property(float)
+    def userLowerLimit(self):
+        """
+        The user-defined lower limit for the spinbox.
+        Returns
+        -------
+        float
+        """
+        return self._user_lower_limit
+
+    @userLowerLimit.setter
+    def userLowerLimit(self, value):
+        """
+        The user-defined lower limit for the spinbox.
+        Parameters
+        ----------
+        value : float
+            The new lower limit value.
+        """
+        self._user_lower_limit = float(value)
+        self._user_upper_limit = max(
+            self._user_upper_limit, self._user_lower_limit)
```

Comment:
@hhslepicka, I would like to know your opinion about two points here:
 1) I always set the attributes _user_lower_limit regardless of the state of _limits_from_channel. I noticed that this is not being done at PyDMSlider and PyDMScale widgets. Do you think this is appropriate?
 2) I check if the lower value being set is larger than the current maximum and move the maximum to leave the range valid. I saw that this is what QSpinBox does. I also noticed that this is not done in PyDMSlider and PyDMScale.